### PR TITLE
Quality of life improvements.

### PR DESCRIPTION
**Pull request by ncareau**
I propose a few changes that simplify and I think help the project.

- When using no contract, use the first available and warn user. (Before, when no contract specified, the app crashed.)
- Why not use only one dockerfile ? I've added the mqtt output in the same dockerfile and moved to the root. In the long run, I think having only one docker will be easier for users. 
- Added a way to use environment variables instead of cli. This is the way to go for docker/k8s as environment are the principal way to configure an app.
- Added .vscode to gitignore.

I'm open to changes, deletion or addition. Just let me know what you think. 

I think that the docker changes will need changes to your .gitlab-ci file. Are you planning to move to gitlab ? or just use it for the registry ?

-NMC

### CODE REVIEW

**Comment by titilambert** [comment]
BIG THANKS
This is a huge one !

File: pyhydroquebec/__main__.py [hunk 1]
```diff
@@ -96,22 +100,42 @@ def main():
         print(VERSION)
         return 0
 
-    if not args.username or not args.password:
+    # Check input for Username, Password and Contract - CLI overwrite ENV variable
+    hydro_user = hydro_pass = hydro_contract = None
+    #Check Env
+    if "PYHQ_USER" in os.environ:
+        hydro_user = os.environ["PYHQ_USER"]
```
Comment (titilambert) [inline pyhydroquebec/__main__.py]: Can be replaced by `hydro_user = os.environ.get("PYHQ_USER")`
If the key is not there, the value will be None.

File: pyhydroquebec/__main__.py [hunk 2]
```diff
@@ -96,22 +100,42 @@ def main():
         print(VERSION)
         return 0
 
-    if not args.username or not args.password:
+    # Check input for Username, Password and Contract - CLI overwrite ENV variable
+    hydro_user = hydro_pass = hydro_contract = None
```
Comment (titilambert) [inline pyhydroquebec/__main__.py]: If you use `os.environ.get...` things this line can be removed

File: pyhydroquebec/__main__.py [hunk 3]
```diff
@@ -96,22 +100,42 @@ def main():
...
+    #Check Env
+    if "PYHQ_USER" in os.environ:
+        hydro_user = os.environ["PYHQ_USER"]
+
+    if "PYHQ_PASSWORD" in os.environ:
+        hydro_pass = os.environ["PYHQ_PASSWORD"]
```
Comment (titilambert) [inline pyhydroquebec/__main__.py]: Same here

File: pyhydroquebec/__main__.py [hunk 4]
```diff
@@ -96,22 +100,42 @@ def main():
...
+
+    if "PYHQ_CONTRACT" in os.environ:
+        hydro_contract = os.environ["PYHQ_CONTRACT"]
```
Comment (titilambert) [inline pyhydroquebec/__main__.py]: Same here

File: pyhydroquebec/__main__.py [hunk 5]
```diff
@@ -96,22 +100,42 @@ def main():
...
+    if args.password:
+        hydro_pass = args.password
+    if args.contract:
+        hydro_contract = args.contract
+
+    if not hydro_user or not hydro_pass:
         parser.print_usage()
         print("pyhydroquebec: error: the following arguments are required: "
-              "-u/--username, -p/--password")
+                "-u/--username, -p/--password")
```
Comment (titilambert) [inline pyhydroquebec/__main__.py]: Keep it aligned.
The `tox -e pylint,lint` command should highlight this issue

File: pyhydroquebec/__main__.py (hunk 1 above)
Comment (ncareau) [reply pyhydroquebec/__main__.py]: Done

File: pyhydroquebec/__main__.py (hunk 2 above)
Comment (ncareau) [reply pyhydroquebec/__main__.py]: Done

File: pyhydroquebec/__main__.py (hunk 4 above)
Comment (ncareau) [reply pyhydroquebec/__main__.py]: Done

File: pyhydroquebec/__main__.py (hunk 5 above)
Comment (ncareau) [reply pyhydroquebec/__main__.py]: Done

File: pyhydroquebec/__main__.py (hunk 3 above)
Comment (ncareau) [reply pyhydroquebec/__main__.py]: Done

**Comment by ncareau** [comment]
Updated the PR. 

I've removed the vscode line in gitignore, you are right that this should be in my system gitignore file.

Also note that github has a registry (in beta) and multiple free CI tools that works well with it (ie. travix-ci). You can even plug the lint step in it to block PR from merging if the lint returns false.

There is also using docker hub (which is currently the defacto for docker container repository) which has automatic build of images. You can probably setup something in less than 10 min without the need for script (ie, the gitlab-ci.yml file) as it uses the Dockerfile directly. Let me know if you need help setting that up. 

Even after all that gitlab is a great option too, i use it a lot for all my private projects.

**Pull request merged**